### ISSUE TITLE
"resizeToAvoidBottomPadding: true" displaces more than it should

### ISSUE THREAD

**Issue by silverAndroid** [flutter/flutter#13475]
## Steps to Reproduce

* Tap the TextField
* You'll notice the entire page is pushed up when the soft keyboard shows up

Gist: https://gist.github.com/silverAndroid/eb40f5faaa2d8a0296baf2f9c337ec2b

![](https://thumbs.gfycat.com/PositiveImpeccableEquestrian-max-14mb.gif)

## Logs

No exception.

## Flutter Doctor
```
[√] Flutter (on Microsoft Windows [Version 10.0.15063], locale en-US, channel master)
    • Flutter at D:\dev\SDKs\flutter
    • Framework revision 51ab5f36c6 (14 hours ago), 2017-12-08 20:39:56 -0800
    • Engine revision edb0201fa2
    • Tools Dart version 1.25.0-dev.11.0
    • Engine Dart version 2.0.0-edge.a38ac7cf127f4611c873c2f2d523c06ce06b1342

[√] Android toolchain - develop for Android devices (Android SDK 26.0.2)
    • Android SDK at D:\dev\SDKs\android-sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-26, build-tools 26.0.2
    • ANDROID_HOME = D:\dev\SDKs\android-sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b01)

[√] Android Studio (version 3.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-915-b01)

[√] IntelliJ IDEA Ultimate Edition (version 2017.3)
    • Flutter plugin version 20.0.3
    • Dart plugin version 173.3727.108

[√] Connected devices
    • LG H915 • LGH9157e1e3243 • android-arm • Android 7.0 (API 24)
```

**Comment by Hixie** [comment]
cc @cbracken might be fallout from your recent work

**Comment by cbracken** [comment]
@silverAndroid Thanks for reporting!

@Hixie Yep - that looks to be the case. This should be a small patch on the Android side to ensure the keyboard is no longer triggering padding, just the inset. Will take a look.

**Comment by pulyaevskiy** [comment]
This seems to be the case for iOS as well. Just updated to latest master and got this:

| Keyboard hidden  | Keyboard visible |
| ------------- | ------------- |
| ![image](https://user-images.githubusercontent.com/441756/34496343-103c73ae-efae-11e7-80bc-07d302060f1e.png)  |  ![image](https://user-images.githubusercontent.com/441756/34496131-34f750a2-efad-11e7-8e45-38686e997689.png)  |

**Comment by hundeva** [comment]
Until this is fixed, a workaround on android is to remove `android:windowSoftInputMode="adjustResize"` from the manifest. This is a behaviour change, but at least you can use the textfields, without blocking the whole screen.

**Comment by ride4sun** [comment]
This is really blocking us to have a first test release of our app to preliminary customers. Is there any way to give that more priority?

**Comment by Hixie** [comment]
I don't quite understand the bug here, can you elaborate?

**Comment by ride4sun** [comment]
The defect  behaves pretty much like silverAndroid  described it above. The content jumps up when the keyboard comes up and bag down when it gets dismissed I tried t counteract the offset by using detecting if the screensize  (_currentScreenSize ) is changing but without much luck:

```
  AppContext createMaterialApp(BuildContext context) => _appContext ??=
          new AppContext(new LayoutBuilder(builder: (context, constraints) {
        _initialScreenSize ??=
            new Size(constraints.maxHeight, constraints.maxWidth);
        //set on every update
        _currentScreenSize =
            new Size(constraints.maxHeight, constraints.maxWidth);
        _log.finest('Current Screen Size: $constraints');
```

**Comment by Hixie** [comment]
The screen doesn't change size when the keyboard appears, so it shouldn't be causing any effect unless you're using SafeArea widgets (which automatically set padding to avoid the keyboard) or Scaffold widgets (which do the same thing unless you set `resizeToAvoidBottomPadding` to false). The original gist uses a Scaffold; setting `resizeToAvoidBottomPadding` to false on that should change to the behavior you want, if I'm understanding correctly.

You can see what the current padding information is using [MediaQuery], but that's only if you do want things to change size.

**Comment by pulyaevskiy** [comment]
> setting resizeToAvoidBottomPadding to false on that should change to the behavior you want

This helped in my case. However, what's the use case for `resizeToAvoidBottomPadding: true` now than? And, when it's set to `true` - is it expected that `bottomNavigationBar` jumps that high with the keyboard open (see my screenshot from above, looks like `keyboardHeight * 2`)?

**Comment by ride4sun** [comment]
resizeToAvoidBottomPadding  default value is not changed in my code. So if I understand u correctly the screen is just moved up and part of it is out of the visible screen are. That's fine with me (Which I did not understand previously). That explains also that my code to detect the keyboard does not work. Anyhow - all this does still not explain why the text is jumping up as visible in the original description of this defect. I agree with @pulyaevskiy it looks like double the height of the keyboard.

**Comment by Hixie** [comment]
For logistical reasons, please file separate bugs for each question.

> what's the use case for resizeToAvoidBottomPadding: true

It's normally what you want. If your main body contents is a list, for example, you wouldn't want the "bottom" of the list to be below the keyboard, because it wouldn't let you see the bottom of the list while the keyboard was up. Similarly if the UI is flexible, you would want it to squeeze while the keyboard is up, so that you can see everything.

> double the height of the keyboard

The jumping up higher than the keyboard isn't expected. Can you confirm that this what this issue is supposed to be about?

**Comment by ride4sun** [comment]
> The jumping up higher than the keyboard isn't expected. Can you confirm that this what this issue is 
supposed

Confirmed

**Comment by Hixie** [comment]
The gist in the original comment of this issue works fine for me. Are you testing alpha or master? If you're on alpha I recommend trying master (or waiting a few hours, I plan to roll alpha tomorrow).

**Comment by ride4sun** [comment]
I am on alpha

**Comment by MrIceman** [comment]
That bug was the reason for me to stop doing Flutter dev for now and switch to C++ Cross Platform..

**Comment by cbracken** [comment]
For the Material side. Fixed by #14406. If you're still seeing issues, please let me know.

We'll need to land a similar patch to fix the Cupertino-specific issue reported by @pulyaevskiy. I've opened #14414 to track that bug.
